### PR TITLE
Add new management sample

### PR DESCRIPTION
**Pull request by Y-Sindo**

### CODE REVIEW

File: samples/Management/ManagementServer/HostedServices/AzureSignalR.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Threading;
```
Comment (wanlwanl) [inline samples/Management/ManagementServer/HostedServices/AzureSignalR.cs:1]: // Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

File: samples/Management/ManagementServer/HostedServices/AzureSignalR.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            ServiceHubContext = await new ServiceHubContextBuilder()
+                .WithConfiguration(_configuration)
+                .WithLoggerFactory(_loggerFactory)
+                .CreateAsync("ManagementSampleHub", cancellationToken);
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
```
Comment (wanlwanl) [inline samples/Management/ManagementServer/HostedServices/AzureSignalR.cs:31]: cancellationToken can be removed?

File: samples/Management/ManagementServer/HostedServices/AzureSignalR.cs (hunk 1 above)
Comment (wanlwanl) [reply samples/Management/ManagementServer/HostedServices/AzureSignalR.cs:1]: also check files don't contain licenses

File: samples/Management/SignalRClient/README.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# SignalR Client
+
+This sample shows how to use SignalR clients to connect Azure SignalR Service without using a web server that host a SignalR hub.
+
+## Build from Scratch
```
Comment (wanlwanl) [inline samples/Management/SignalRClient/README.md:5]: from -> From

File: samples/Management/SignalRClient/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+### Add Management SDK to your project
+
+```
+dotnet add package Microsoft.Azure.SignalR.Management -v 1.0.0-*
```
Comment (wanlwanl) [inline samples/Management/SignalRClient/README.md:10]: -v 1.0.0-* is too old?

File: samples/Management/SignalRClient/README.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+### Handle connection closed event
+
+Sometimes SignalR clients may be disconnected by Azure SignalR Service, the `Closed` event handler will be useful to figure out the reason.
+
+```C#
+connection.Closed += async ex =>
+{
+	// handle exception here
```
Comment (wanlwanl) [inline samples/Management/SignalRClient/README.md:27]: Indentation

File: samples/Management/SignalRClient/README.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+You can stop the client connection anytime you want.
+
+```C#
+await connection.StopAsync();
+```
+
+## Full Sample
+
+The full message publisher sample can be found [here](.). The usage of this sample can be found [here](<https://github.com/aspnet/AzureSignalR-samples/tree/master/samples/Management#start-signalr-clients>).
```
Comment (wanlwanl) [inline samples/Management/SignalRClient/README.md:60]: links are out-of-date?

File: samples/Management/ManagementServer/Controllers/NegotiateController.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.SignalR.Management;
+
+namespace ManagementServer
+{
+    [ApiController]
+    public class NegotiateController : ControllerBase
```
Comment (wanlwanl) [inline samples/Management/ManagementServer/Controllers/NegotiateController.cs:12]: NegotiateController --> NegotiationController

File: samples/Management/ManagementServer/ManagementServer.csproj [hunk 8]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+    <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>
+    <UserSecretsId>ManagementSample</UserSecretsId>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Azure.SignalR.Management" Version="1.9.3-preview1-10802" />
```
Comment (wanlwanl) [inline samples/Management/ManagementServer/ManagementServer.csproj:10]: is it possible to use stable version?

File: samples/Management/SignalRClient/README.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
+# SignalR Client
```
Comment (wanlwanl) [inline samples/Management/SignalRClient/README.md:1]: is there any readme for negotiation controller and management server?

File: samples/Management/ManagementServer/HostedServices/AzureSignalR.cs (hunk 2 above)
Comment (Y-Sindo) [reply samples/Management/ManagementServer/HostedServices/AzureSignalR.cs:31]: This is an interface implementation.

File: samples/Management/SignalRClient/README.md (hunk 9 above)
Comment (Y-Sindo) [reply samples/Management/SignalRClient/README.md:1]: added

File: samples/Management/ManagementServer/ManagementServer.csproj (hunk 8 above)
Comment (Y-Sindo) [reply samples/Management/ManagementServer/ManagementServer.csproj]: Now use 1.10.0

File: samples/Management/LegacySample/README.md [hunk 10]
```diff
@@ -1,70 +0,0 @@
-Azure SignalR Service Management SDK Sample (Legacy)
```
Comment (wanlwanl) [inline samples/Management/LegacySample/README.md:1]: >Azure [](start = 0, length = 5)

why delete the legacy readme?

Comment (Y-Sindo) [reply samples/Management/LegacySample/README.md:1]: We don't want user to use the legacy APIs, so the legacy sample is deleted.

File: samples/Management/MessagePublisher/MessagePublisher.csproj [hunk 11]
```diff
@@ -2,13 +2,13 @@
...
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net5</TargetFramework>
     <UserSecretsId>ManagementSample</UserSecretsId>
     <RootNamespace>Microsoft.Azure.SignalR.Samples.Management</RootNamespace>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.SignalR.Management" Version="1.*" />
+    <PackageReference Include="Microsoft.Azure.SignalR.Management" Version="1.10.*" />
```
Comment (wanlwanl) [inline samples/Management/MessagePublisher/MessagePublisher.csproj:11]: >10 [](start = 78, length = 2)

why change it?

Comment (Y-Sindo) [reply samples/Management/MessagePublisher/MessagePublisher.csproj:11]: The new sample needs package version 1.10.*.

File: samples/Management/NegotiationServer/Controllers/NegotiateController.cs [hunk 12]
```diff
@@ -2,37 +2,36 @@
...
-            var connectionString = configuration["Azure:SignalR:ConnectionString"];
-            _serviceManager = new ServiceManagerBuilder()
-                .WithOptions(o => o.ConnectionString = connectionString)
-                .Build();
+            _hubContext = signalrService.HubContext;
         }
 
-        [HttpPost("{hub}/negotiate")]
-        public ActionResult Index(string hub, string user)
+        [HttpPost("negotiate")]
```
Comment (wanlwanl) [inline samples/Management/NegotiationServer/Controllers/NegotiateController.cs:21]: >negotiate [](start = 19, length = 9)

emmmm, some cx has multiple hubs in production, could you consider it?

**Comment by wanlwanl** [comment]
>             var connection = new HubConnectionBuilder().WithUrl(url).Build();

url = <baseEndpoint>/<hub> is more nature

---
Refers to: samples/Management/SignalRClient/Program.cs:52 in 6b4646f. [](commit_id = 6b4646f8da78455f32775060e87a8df06a2f0a9a, deletion_comment = False)

File: samples/Management/MessagePublisher/MessagePublisher.csproj (hunk 11 above)
Comment (wanlwanl) [reply samples/Management/MessagePublisher/MessagePublisher.csproj:11]: 1.* will not use the latest start with 1.?

File: samples/Management/LegacySample/README.md (hunk 10 above)
Comment (wanlwanl) [reply samples/Management/LegacySample/README.md:1]: why not also delete the code?

File: samples/Management/NegotiationServer/Controllers/NegotiateController.cs (hunk 12 above)
Comment (Y-Sindo) [reply samples/Management/NegotiationServer/Controllers/NegotiateController.cs]: Will add a strongly typed hub after it is ready.

**Review by vicancy — APPROVED** [review]

File: samples/Management/NegotiationServer/Controllers/NegotiateController.cs [hunk 13]
```diff
@@ -18,7 +18,7 @@ public NegotiateController(SignalRService signalrService)
             _hubContext = signalrService.HubContext;
         }
 
-        [HttpPost("negotiate")]
+        [HttpPost("ManagementSampleHub/negotiate")]
```
Comment (vicancy) [inline samples/Management/NegotiationServer/Controllers/NegotiateController.cs:21]: why change?

File: samples/Management/MessagePublisher/README.md [hunk 14]
```diff
@@ -8,25 +8,26 @@ This sample shows how to use [Microsoft.Azure.SignalR.Management](https://www.nu
 ### Add Management SDK to your project
 
 ```
-dotnet add package Microsoft.Azure.SignalR.Management -v 1.0.0-*
+dotnet add package Microsoft.Azure.SignalR.Management -v 1.10.*
```
Comment (vicancy) [inline samples/Management/MessagePublisher/README.md:11]: this is not consistent with the project file

File: samples/Management/NegotiationServer/Controllers/NegotiateController.cs (hunk 13 above)
Comment (vicancy) [reply samples/Management/NegotiationServer/Controllers/NegotiateController.cs:21]: What's the story now if I have multiple hubs

File: samples/Management/NegotiationServer/SignalRService.cs [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.SignalR.Management;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace NegotiationServer
+{
+    public class SignalRService : IHostedService
```
Comment (vicancy) [inline samples/Management/NegotiationServer/SignalRService.cs:13]: So the recommended way is every Hub has a service?

File: samples/Management/NegotiationServer/Startup.cs [hunk 16]
```diff
@@ -22,6 +22,9 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddControllers();
+            services
+                .AddSingleton<SignalRService>()
```
Comment (vicancy) [inline samples/Management/NegotiationServer/Startup.cs:26]: This looks a little bit tricky to DI hostedservice as a recommended way in our sample...

**Review by vicancy — APPROVED** [review]

**Pull request merged**